src/main/java/address/sync/CloudService.java

```diff
@@ -0,0 +1,448 @@
+package address.sync;
+
+import address.model.datatypes.Tag;
+import address.model.datatypes.Person;
+import address.sync.model.CloudTag;
+import address.sync.model.CloudPerson;
+import address.util.JsonUtil;
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.*;
+
+/**
+ * Emulates the local cloud service
+ *
+ * This component is responsible for providing a high-level API for communication with the cloud,
+ * as well as the reformatting the cloud's response into a format understood by the local logic.
+ *
+ * Most of the responses returned should include the rate limit status if the respective cloud response(s)
+ * contain(s) them
+ */
+public class CloudService implements ICloudService {
+    private static final int RESOURCES_PER_PAGE = 100;
+
+    private final CloudSimulator cloud;
+
+    CloudService(boolean shouldSimulateUnreliableNetwork) {
+        cloud = new CloudSimulator(shouldSimulateUnreliableNetwork);
+    }
+
+    /**
+     * Checks whether a response from the cloud is valid
+     *
+     * A response is considered valid if the request has successfully executed
+     * This does not include the case 304 where it has the same return result as before
+     *
+     * @param response
+     * @return
+     */
+    public static boolean isValid(RawCloudResponse response) {
+        switch (response.getResponseCode()) {
+            case 200:
+            case 201:
+            case 202:
+            case 203:
+            case 204:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * Gets the list of persons for addressBookName, if quota is available
+     *
+     * Consumes 1 + floor(persons size/RESOURCES_PER_PAGE) API usage
+     * @param addressBookName
+     * @return wrapped response with list of persons
+     * @throws IOException if content cannot be interpreted
+     */
+    @Override
+    public ExtractedCloudResponse<List<Person>> getPersons(String addressBookName) throws IOException {
+        int curPageNumber = 1;
+        RawCloudResponse cloudResponse;
+        List<CloudPerson> cloudPersons = new ArrayList<>();
+        do {
+            cloudResponse = cloud.getPersons(addressBookName, curPageNumber, RESOURCES_PER_PAGE, null);
+            cloudPersons.addAll(getDataListFromBody(cloudResponse.getBody(), CloudPerson.class));
```

Comment:
Haven't considered that case. The loop should end early with an empty response, I think.

Will resolve that in a new PR.